setup.py

```diff
@@ -230,7 +230,7 @@ def run(self):
                     "libmigraphx_tf.so.2",
                 ]
 
-                tensorrt_dependencies = ["libnvinfer.so.8", "libnvinfer_plugin.so.8", "libnvonnxparser.so.8"]
+                tensorrt_dependencies = ["libnvinfer.so.10", "libnvinfer_plugin.so.10", "libnvonnxparser.so.10"]
```

Comment:
+@yifanl fyi

Reply:
do we need this? is there a symlink from lbnvinfer.so to libnvinfer.so.8 ? would we be able add libnvinfer.so ?

Reply:
did you actually hit an issue with this?
i'm actually not sure we need this stuff anymore. wasn't it used in the old days to work around auditwheel and ensure it doesn't try to copy these dependencies into the whl?

Reply:
Each TRT does have a symlink:
i.e For TRT8.6 installed in Ubuntu:
![image](https://github.com/microsoft/onnxruntime/assets/109183385/a26fca7b-09db-4661-bf1c-6e7b91a23137)
After installing TRT10, libnvinfer.so was re-linked to libnvinfer.so.10:
![image](https://github.com/microsoft/onnxruntime/assets/109183385/3e6b4385-6324-4e7e-a5a8-7a9680767c85)

Reply:
ok this change is required. i downloaded whl package and see the libnvinfer libs copied in there under ort-nightly-gpu.libs